### ISSUE TITLE
Can't use `PimpedNodeList` on `NodeListOf`

### ISSUE THREAD

**Issue by avernet** [scala-js/scala-js-dom#388]
Following the instructions on [From ES6 to Scala: Advanced](https://www.scala-js.org/doc/sjs-for-js/es6-to-scala-part3.html), I do:

```scala
val images = dom.document.querySelectorAll("img").asInstanceOf[NodeListOf[HTMLImageElement]]
images.map(_.src)
```

The second line doesn't compile, and the compiler tells us: `value map is not a member of org.scalajs.dom.NodeListOf[org.scalajs.dom.raw.HTMLImageElement]`. If I `import org.scalajs.dom.ext._`, I'd expect my `NodeListOf` to be "pimped", but apparently `PimpedNodeList` only applies on `NodeList`, and not on `NodeListOf`. You can try this in this [scastie](https://scastie.scala-lang.org/buywMTHERNe7gKvGA2dmVg).

Am I missing something, or is there something incorrect in the documentation? (Or both!)

**Comment by mseddon** [comment]
@sjrd do we still want things named PimpedAnything? Perhaps EnrichedNodeList and keep it buried ;)

**Comment by sjrd** [comment]
I think the current naming convention is `NodeListOps`.

**Comment by japgolly** [comment]
This was implemented in #458
